src/main/java/frc/robot/util/DrawMechanism.java

```diff
@@ -0,0 +1,278 @@
+        MechanismLigament2d bumper1 = new MechanismLigament2d("bumper1", 13, 0, 4, colorBumper);
+        MechanismLigament2d bumper2 = new MechanismLigament2d("bumper2", 13, 180, 4, colorBumper);
+        m_root.append(bumper1);
+        m_root.append(bumper2);
+
+        // make the elevator
+        // elevator is 10 inches high and 12 inches back
+        // go back 12 inches
+        MechanismLigament2d spaceElevatorBack = new MechanismLigament2d("spaceEL", 12, 180, 0.2, colorSpace);
+        m_root.append(spaceElevatorBack);
+        // go up 10 inches
+        MechanismLigament2d spaceElevatorUp = new MechanismLigament2d("elUp", 10, -90, 0.25, colorSpace);
+        spaceElevatorBack.append(spaceElevatorUp);
+        // the elevator is slanted at 55 degrees.
+        m_elevator = new MechanismLigament2d("elevator", 50, 55.0-90.0, 3.0, colorElevator);
+        
+        // elevator bottom stage is permanently there so appending it as a non-moving ligament. m_elevator will be the part that moves.
+        MechanismLigament2d elevatorBar = new MechanismLigament2d("elevatorBar", 50, 55.0-90.0, 3.0, colorElevator);
+        spaceElevatorUp.append(m_elevator);
+        spaceElevatorUp.append(elevatorBar);
+
+        // make the wrist
+        // wristHandle is portion that is horizontal and extends out from elevator carriage
+        MechanismLigament2d wristHandle = new MechanismLigament2d("wristHandle", 17, -55, 3.0, colorWrist);
+        m_elevator.append(wristHandle);
+        // m_wrist is moving part of the wrist (angle is read by the absolute encoder)
+        m_wrist = new MechanismLigament2d("wrist", 3.5, 150.0, 3.0, colorWrist);
+        wristHandle.append(m_wrist);
+
+        // make the intake
+        m_intake = new MechanismLigament2d("intake", 13, 0, 4.0, colorIntakeDisabled);
+        m_wrist.append(m_intake);
+
+        // set the wrist and intake angles to be consistent
+        setWristAngle(WristConstants.kStowPos);
+
+        // put the Mechanism2D on the dashboard
+        SmartDashboard.putData("Mech2d", m_mech2d);
+
+        // We have the Mech2d, so put some commands on the SmartDashboard
+        addCommands(this);
+    }
+
+    private void addCommands(DrawMechanism mechanism) {
+        // some commands to move the Mech2d diagram
+        SmartDashboard.putData("wrist in", new InstantCommand(() -> mechanism.setWristAngle(WristConstants.kStowPos)));
+        SmartDashboard.putData("wrist out", new InstantCommand(() -> mechanism.setWristAngle(WristConstants.kIntakeCubePos)));
+        SmartDashboard.putData("elevator down", new InstantCommand(() -> mechanism.setElevatorExtension(0.1)));
+        SmartDashboard.putData("elevator up", new InstantCommand(() -> mechanism.setElevatorExtension(1.3)));
+        SmartDashboard.putData("drivetrain in", new InstantCommand(() -> mechanism.setDistanceToGrid(16.5)));
+        SmartDashboard.putData("drivetrain away", new InstantCommand(() -> mechanism.setElevatorExtension(48)));
```

Comment:
"drivetrain away" sets elevator extension to 48 meters

Reply:
right, no more late night coding...